### PR TITLE
[RFR] Heatmap

### PR DESCRIPTION
**Pull request by ThieryMichel**
capture du 2018-01-02 17-29-59

### CODE REVIEW

File: src/app/js/formats/heatmap/HeatMapView.js [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+                                            className={css(styles.rotateSpan)}
+                                        >
+                                            {xKey}
+                                        </span>
+                                    </div>
+                                </td>
+                                {yAxis.map(yKey => (
+                                    <td
+                                        className={css(styles.td)}
+                                        key={yKey}
```
Comment (fzaninotto) [inline src/app/js/formats/heatmap/HeatMapView.js]: not good, you'll end up with duplicate keys as it's repeated for every line. I advise you to use a compound of `xKey` and `yKey` here

File: src/app/js/formats/heatmap/parseChartData.spec.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
+import expect from 'expect';
+
+import { mapSourceToX, mapTargetToX } from './parseChartData';
+
+describe.only('parseChartData', () => {
```
Comment (fzaninotto) [inline src/app/js/formats/heatmap/parseChartData.spec.js]: `.only` spotted!

**Review by fzaninotto — CHANGES_REQUESTED** [review]

**Comment by fzaninotto** [comment]
Seing the screenshot, I worry about the horizontal scroll bar (which is never a good ui). Can't the columns width adjust to the available width?

**Comment by ThieryMichel** [comment]
If we make the columns smaller we won't have enough place for the label, also we can flip the two axes, it's in the option of the format. Granted it won't fix all case.

capture du 2018-01-03 18-24-12

File: src/app/js/formats/heatmap/HeatMapView.js (hunk 1 above)
Comment (ThieryMichel) [reply src/app/js/formats/heatmap/HeatMapView.js]: I thought that as long as all the key in the array of component are unique, it's good, even if there is an element with the same key elsewhere. Also react print no duplicate key error.
I guess it cannot hurt, just not sure it is really needed.

**Comment by fzaninotto** [comment]
I think it makes more sense with the years as columns. what do the PO say?

**Review by fzaninotto — APPROVED** [review]

**Pull request merged**